ste/jobStatusManager.go

```diff
@@ -61,14 +61,28 @@ func (jm *jobMgr) statusMgrClosed() bool {
 
 /* These functions should not fail */
 func (jm *jobMgr) SendJobPartCreatedMsg(msg JobPartCreatedMsg) {
-	jm.jstm.partCreated <- msg
-	if msg.IsFinalPart {
-		// Inform statusManager that this is all parts we've
-		close(jm.jstm.partCreated)
+	defer func() {
```

Comment:
is there an easy test that we can add to our repo that reproduces the bug

Reply:
The main issue with this bug is that it's a bit arbitrary and non-deterministic to reproduce. It is all about the timing of when the send was made on a channel that has already been closed. This doesn't always happen. But in the case, it does in the future, we have some safeguards around that.